### ISSUE TITLE
Consider adding nmap/v2.Option

### ISSUE THREAD

**Issue by SpikesDivZero** [Ullaakut/nmap#70]
It would be a minor breaking change, but would it be worth adding something like the below type?

```
type Option func(*Scanner)
```

The use case I'm seeing is two-fold:
* Better grouping of options on pkg.go.dev (for example, see `type CallOption` on [grpc's Index](https://pkg.go.dev/google.golang.org/grpc#pkg-index))
* Making it a touch easier for downstream users to maintain a set of base options.

For the latter case, I'm currently doing something like so:

```
type X struct {
	baseOptions []func(*nmap.Scanner)
}
func newX() X {
	return X{
		baseOptions: []func(*nmap.Scanner){
			nmap.With1(),
			nmap.With2(),
			nmap.With3(),
		},
	}
}
func doTheThing(host string) {
	s, _ := nmap.NewScanner(ne.baseOptions...)
	scanner.AddOptions(nmap.WithTargets(host))
	if isIP6(host) {
		scanner.AddOptions(nmap.WithIPv6Scanning())
	}
```

The key improvement to the above code would be mainly in clarity, as the type a user is writing out is a bit less abstract, and it doesn't require a later reader to go cross-reference the docs to verify its accuracy.

Thanks for the time and consideration.

**Comment by Ullaakut** [comment]
Sounds like a good idea, and should not be too difficult to implement :)

**Comment by Ullaakut** [comment]
Fixed by #71 

@SpikesDivZero don't hesitate to let us know if that is satisfactory :)

**Comment by SpikesDivZero** [comment]
Apologies for the slow reply.

I'd tested this via `go get github.com/Ullaakut/nmap/v2@ac12e8662712` and it looks to work splendidly. It was indeed a breaking change, as was to be expected, but the fix was relatively minor.

```
-       baseOptions []func(*nmap.Scanner)
+       baseOptions []nmap.Option
...
-       ne.baseOptions = []func(*nmap.Scanner){
+       ne.baseOptions = []nmap.Option{
```

Thanks so much!
